lib/Raven/Client.php

```diff
@@ -559,9 +559,22 @@ protected function get_http_data()
         if (!empty($_POST)) {
             $result['data'] = $_POST;
         }
+        
+        // if the $_POST super global is empty, try to retrieve a json string
+        // prior to PHP 5.6, a stream opened with php://input could only be read once, so only do this with 5.6 or up
+        if (version_compare(PHP_VERSION, '5.6', '>=') && empty($_POST)) {
+            try {
+                $data = file_get_contents('php://input');
+                $result['data'] = json_decode($data, true);
```

Comment:
Thinking more about this, I think theres two more concerns we should address:

1. Dont read the entire stream (I'm not sure if this can be a DoS vector). This seems achievable via the $maxlen param, but maybe can also be achieved with one of the other file helpers.
2. It'd probably be good that we still capture the raw value even if its not JSON. Not really a requirement just a nice to have.